plugins/googlesignin/class.googlesignin.plugin.php

```diff
@@ -0,0 +1,124 @@
+ * @license Proprietary
+ */
+
+
+/**
+ * Class GoogleSignInPlugin
+ *
+ * Expose the functionality of the core class Gdn_OAuth2 to create SSO workflows.
+ */
+
+class GoogleSignInPlugin extends Gdn_OAuth2 {
+    CONST AUTHORIZEURL = 'https://accounts.google.com/o/oauth2/v2/auth';
+    CONST TOKENURL = 'https://oauth2.googleapis.com/token';
+    CONST PROFILEURL = 'https://openidconnect.googleapis.com/v1/userinfo';
+    CONST PROFILENAME = 'name';
+    CONST ACCEPTEDSCOPE = 'email openid profile';
+    /**
+     * Set the key for saving OAuth settings in GDN_UserAuthenticationProvider
+     */
+    public function __construct() {
+        $this->setProviderKey('googlesignin');
+        $this->settingsView = 'plugins/settings/googlesignin';
+    }
+
+    /**
+     *  Return all the information saved in provider table, add hardcoded values.
+     *
+     * @return array Stored provider data (secret, client_id, etc.).
+     */
+    public function provider() {
+        if (!$this->provider) {
+            $this->provider = Gdn_AuthenticationProviderModel::getProviderByKey($this->providerKey);
+
+            // These URLs are added here instead of being stored in the DB in case they ever change, we will not have to update the DB.
+            $this->provider['AuthorizeUrl'] = self::AUTHORIZEURL;
+            $this->provider['TokenUrl'] = self::TOKENURL;
+            $this->provider['ProfileUrl'] = self::PROFILEURL;
+
+            // Scope
+            $this->provider['AcceptedScope'] = self::ACCEPTEDSCOPE;
+
+            // Translate claims coming back from Google.
+            $this->provider['ProfileKeyName'] = self::PROFILENAME;
+            $this->provider['AcceptedScope'] = 'email openid profile';
+
+            $this->provider['ProfileKeyUniqueID'] = 'sub';
+            $this->provider['ProfileKeyFullName'] = null;
+
+            // provider Name is what puts the text on the button and determines the CSS class.
+            $this->provider['Name'] = 'Google';
+        }
+
+        return $this->provider;
+    }
+
+    /**
+     * Create a controller to deal with plugin settings in dashboard.
+     *
+     * @param Gdn_Controller $sender.
+     * @param Gdn_Controller $args.
+     */
+    public function settingsEndpoint($sender, $args) {
+        $sender->permission('Garden.Settings.Manage');
+        $model = new Gdn_AuthenticationProviderModel();
+
+        /* @var Gdn_Form $form */
+        $form = new Gdn_Form();
+        $form->setModel($model);
+        $sender->Form = $form;
+
+        if (!$form->authenticatedPostBack()) {
+            $provider = $this->provider();
+            $form->setData($provider);
+        } else {
+
+            $form->setFormValue('AuthenticationKey', $this->getProviderKey());
+
+            $sender->Form->validateRule('AssociationKey', 'ValidateRequired', 'You must provide a unique AccountID.');
+            $sender->Form->validateRule('AssociationSecret', 'ValidateRequired', 'You must provide a Secret');
```

Comment:
Validation messages break the layout with not styled messages and escaped html tags:
<img width="1017" alt="Screen Shot 2019-04-06 at 1 20 46 PM" src="https://user-images.githubusercontent.com/15682507/55672941-498b7080-586f-11e9-81cc-7782aae0be47.png">

If you change twig  line to `raw` format that fix the issue:
```
{{ form.errors() | raw }}
```